<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TestListWidgetToolButton</class>
 <widget class="QMainWindow" name="TestListWidgetToolButton">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>859</width>
    <height>449</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <family>Consolas</family>
    <pointsize>16</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>TestListWidgetToolButton</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QSplitter" name="splitter">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>10</y>
      <width>831</width>
      <height>361</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
    <widget class="QToolBox" name="toolBox">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>240</width>
       <height>0</height>
      </size>
     </property>
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="page_01">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>240</width>
        <height>244</height>
       </rect>
      </property>
      <attribute name="icon">
       <iconset resource="icons.qrc">
        <normaloff>:/icons/410.bmp</normaloff>:/icons/410.bmp</iconset>
      </attribute>
      <attribute name="label">
       <string>List Operation</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="0">
        <widget class="QToolButton" name="TB_05Initialize">
         <property name="minimumSize">
          <size>
           <width>204</width>
           <height>32</height>
          </size>
         </property>
         <property name="text">
          <string>Initialize list</string>
         </property>
         <property name="toolButtonStyle">
          <enum>Qt::ToolButtonTextBesideIcon</enum>
         </property>
         <property name="autoRaise">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QToolButton" name="TB_06ClearList">
         <property name="minimumSize">
          <size>
           <width>204</width>
           <height>32</height>
          </size>
         </property>
         <property name="text">
          <string>Clear list</string>
         </property>
         <property name="toolButtonStyle">
          <enum>Qt::ToolButtonTextBesideIcon</enum>
         </property>
         <property name="autoRaise">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QToolButton" name="TB_07Insert">
         <property name="minimumSize">
          <size>
           <width>204</width>
           <height>32</height>
          </size>
         </property>
         <property name="text">
          <string>Insert item</string>
         </property>
         <property name="toolButtonStyle">
          <enum>Qt::ToolButtonTextBesideIcon</enum>
         </property>
         <property name="autoRaise">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QToolButton" name="TB_08Append">
         <property name="minimumSize">
          <size>
           <width>204</width>
           <height>32</height>
          </size>
         </property>
         <property name="text">
          <string>Append item</string>
         </property>
         <property name="toolButtonStyle">
          <enum>Qt::ToolButtonTextBesideIcon</enum>
         </property>
         <property name="autoRaise">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QToolButton" name="TB_09Delete">
         <property name="minimumSize">
          <size>
           <width>204</width>
           <height>32</height>
          </size>
         </property>
         <property name="text">
          <string>Delete item</string>
         </property>
         <property name="toolButtonStyle">
          <enum>Qt::ToolButtonTextBesideIcon</enum>
         </property>
         <property name="autoRaise">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="page_02">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>240</width>
        <height>244</height>
       </rect>
      </property>
      <attribute name="icon">
       <iconset resource="icons.qrc">
        <normaloff>:/icons/408.bmp</normaloff>:/icons/408.bmp</iconset>
      </attribute>
      <attribute name="label">
       <string>List Sort</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0">
        <widget class="QCheckBox" name="CB_02Sortable">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>200</width>
           <height>32</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>Allow Sorting</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QToolButton" name="TB_10Ascending">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>200</width>
           <height>32</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>Ascending sort</string>
         </property>
         <property name="toolButtonStyle">
          <enum>Qt::ToolButtonTextBesideIcon</enum>
         </property>
         <property name="arrowType">
          <enum>Qt::UpArrow</enum>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QToolButton" name="TB_11Descending">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>200</width>
           <height>32</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>Descending sort</string>
         </property>
         <property name="toolButtonStyle">
          <enum>Qt::ToolButtonTextBesideIcon</enum>
         </property>
         <property name="autoRaise">
          <bool>false</bool>
         </property>
         <property name="arrowType">
          <enum>Qt::DownArrow</enum>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>109</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="page_03">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>472</width>
        <height>230</height>
       </rect>
      </property>
      <attribute name="icon">
       <iconset resource="icons.qrc">
        <normaloff>:/icons/412.bmp</normaloff>:/icons/412.bmp</iconset>
      </attribute>
      <attribute name="label">
       <string>Signal shoting</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_03">
         <property name="spacing">
          <number>6</number>
         </property>
         <item>
          <widget class="QToolButton" name="TB_12ClearText">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>130</width>
             <height>0</height>
            </size>
           </property>
           <property name="text">
            <string>Clear Text</string>
           </property>
           <property name="toolButtonStyle">
            <enum>Qt::ToolButtonTextBesideIcon</enum>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="TB_13AppendSpace">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>120</width>
             <height>0</height>
            </size>
           </property>
           <property name="text">
            <string>Add Space</string>
           </property>
           <property name="toolButtonStyle">
            <enum>Qt::ToolButtonTextBesideIcon</enum>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="CB_03MouseTracking">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>190</width>
             <height>0</height>
            </size>
           </property>
           <property name="text">
            <string>Mouse Tracking</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QPlainTextEdit" name="plainTextEdit">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>454</width>
           <height>0</height>
          </size>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QGroupBox" name="groupBox">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string/>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_01">
        <property name="spacing">
         <number>2</number>
        </property>
        <item>
         <widget class="QLabel" name="label">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>current item variety:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="lineEdit">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="CB_01Editable">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Editable</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_02">
        <property name="spacing">
         <number>0</number>
        </property>
        <item>
         <widget class="QToolButton" name="TB_01SelectItem">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>SelectItem</string>
          </property>
          <property name="popupMode">
           <enum>QToolButton::MenuButtonPopup</enum>
          </property>
          <property name="toolButtonStyle">
           <enum>Qt::ToolButtonTextBesideIcon</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="TB_02SelectAll">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>All</string>
          </property>
          <property name="toolButtonStyle">
           <enum>Qt::ToolButtonTextBesideIcon</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="TB_03SelectNone">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>None</string>
          </property>
          <property name="toolButtonStyle">
           <enum>Qt::ToolButtonTextBesideIcon</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="TB_04SelectInverse">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Inverse</string>
          </property>
          <property name="toolButtonStyle">
           <enum>Qt::ToolButtonTextBesideIcon</enum>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="Line" name="line">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QListWidget" name="listWidget">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="contextMenuPolicy">
         <enum>Qt::CustomContextMenu</enum>
        </property>
        <item>
         <property name="text">
          <string>New Item 01</string>
         </property>
         <property name="font">
          <font>
           <family>Consolas</family>
           <pointsize>16</pointsize>
          </font>
         </property>
         <property name="checkState">
          <enum>Checked</enum>
         </property>
         <property name="icon">
          <iconset resource="icons.qrc">
           <normaloff>:/icons/check2.ico</normaloff>:/icons/check2.ico</iconset>
         </property>
        </item>
        <item>
         <property name="text">
          <string>New Item 02</string>
         </property>
         <property name="font">
          <font>
           <family>Consolas</family>
           <pointsize>16</pointsize>
          </font>
         </property>
         <property name="checkState">
          <enum>Unchecked</enum>
         </property>
         <property name="icon">
          <iconset resource="icons.qrc">
           <normaloff>:/icons/check2.ico</normaloff>:/icons/check2.ico</iconset>
         </property>
        </item>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="iconSize">
    <size>
     <width>32</width>
     <height>32</height>
    </size>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonTextBesideIcon</enum>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actList_Init"/>
   <addaction name="actList_Clear"/>
   <addaction name="actList_Insert"/>
   <addaction name="actList_Append"/>
   <addaction name="actList_Delete"/>
   <addaction name="separator"/>
  </widget>
  <action name="actList_Init">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/icons/128.bmp</normaloff>:/icons/128.bmp</iconset>
   </property>
   <property name="text">
    <string>Initialize_List</string>
   </property>
   <property name="toolTip">
    <string>初始化列表</string>
   </property>
   <property name="font">
    <font>
     <family>Consolas</family>
     <pointsize>16</pointsize>
    </font>
   </property>
   <property name="shortcut">
    <string>Ctrl+I</string>
   </property>
  </action>
  <action name="actList_Clear">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/icons/delete1.bmp</normaloff>:/icons/delete1.bmp</iconset>
   </property>
   <property name="text">
    <string>Clear_List</string>
   </property>
   <property name="toolTip">
    <string>清除列表</string>
   </property>
   <property name="font">
    <font>
     <family>Consolas</family>
     <pointsize>16</pointsize>
    </font>
   </property>
   <property name="shortcut">
    <string>Ctrl+L</string>
   </property>
  </action>
  <action name="actList_Insert">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/icons/424.bmp</normaloff>:/icons/424.bmp</iconset>
   </property>
   <property name="text">
    <string>Insert_Item</string>
   </property>
   <property name="toolTip">
    <string>插入项</string>
   </property>
   <property name="font">
    <font>
     <family>Consolas</family>
     <pointsize>16</pointsize>
    </font>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+I</string>
   </property>
  </action>
  <action name="actList_Append">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/icons/316.bmp</normaloff>:/icons/316.bmp</iconset>
   </property>
   <property name="text">
    <string>Append_Item</string>
   </property>
   <property name="toolTip">
    <string>添加项</string>
   </property>
   <property name="font">
    <font>
     <family>Consolas</family>
     <pointsize>16</pointsize>
    </font>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+A</string>
   </property>
  </action>
  <action name="actList_Delete">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/icons/324.bmp</normaloff>:/icons/324.bmp</iconset>
   </property>
   <property name="text">
    <string>Delete_Item</string>
   </property>
   <property name="toolTip">
    <string>删除当前项</string>
   </property>
   <property name="font">
    <font>
     <family>Consolas</family>
     <pointsize>16</pointsize>
    </font>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+D</string>
   </property>
  </action>
  <action name="actSelect_All">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/icons/122.bmp</normaloff>:/icons/122.bmp</iconset>
   </property>
   <property name="text">
    <string>Select_All</string>
   </property>
   <property name="toolTip">
    <string>全选</string>
   </property>
   <property name="font">
    <font>
     <family>Consolas</family>
     <pointsize>16</pointsize>
    </font>
   </property>
  </action>
  <action name="actSelect_None">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/icons/123.bmp</normaloff>:/icons/123.bmp</iconset>
   </property>
   <property name="text">
    <string>Select_None</string>
   </property>
   <property name="toolTip">
    <string>全不选</string>
   </property>
   <property name="font">
    <font>
     <family>Consolas</family>
     <pointsize>16</pointsize>
    </font>
   </property>
  </action>
  <action name="actSelect_Inverse">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/icons/326.bmp</normaloff>:/icons/326.bmp</iconset>
   </property>
   <property name="text">
    <string>Select_Inverse</string>
   </property>
   <property name="toolTip">
    <string>反选</string>
   </property>
   <property name="font">
    <font>
     <family>Consolas</family>
     <pointsize>16</pointsize>
    </font>
   </property>
  </action>
  <action name="actSelect_PopMenu">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/icons/124.bmp</normaloff>:/icons/124.bmp</iconset>
   </property>
   <property name="text">
    <string>Select_PopMenu</string>
   </property>
   <property name="toolTip">
    <string>项选择</string>
   </property>
   <property name="font">
    <font>
     <family>Consolas</family>
     <pointsize>16</pointsize>
    </font>
   </property>
  </action>
  <action name="actSys_Quit">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/icons/exit.bmp</normaloff>:/icons/exit.bmp</iconset>
   </property>
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="toolTip">
    <string>退出</string>
   </property>
   <property name="font">
    <font>
     <family>Consolas</family>
     <pointsize>16</pointsize>
    </font>
   </property>
  </action>
 </widget>
 <resources>
  <include location="icons.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actSelect_PopMenu</sender>
   <signal>triggered()</signal>
   <receiver>actSelect_Inverse</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actSys_Quit</sender>
   <signal>triggered()</signal>
   <receiver>TestListWidgetToolButton</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>423</x>
     <y>214</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actList_Clear</sender>
   <signal>triggered()</signal>
   <receiver>listWidget</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>541</x>
     <y>289</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>TB_12ClearText</sender>
   <signal>clicked()</signal>
   <receiver>plainTextEdit</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-116</x>
     <y>199</y>
    </hint>
    <hint type="destinationlabel">
     <x>130</x>
     <y>296</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>TB_12ClearText</sender>
   <signal>clicked()</signal>
   <receiver>lineEdit</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-94</x>
     <y>202</y>
    </hint>
    <hint type="destinationlabel">
     <x>562</x>
     <y>91</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
